packages/core/src/logger/apiLogger.ts

```diff
@@ -0,0 +1,260 @@
+import {
+  HttpRequest,
+  HttpResponse,
+  ApiLoggerInterface,
+  LoggerInterface,
+  LoggingOptions,
+  HttpRequestLoggingOptions,
+  HttpMessageLoggingOptions,
+  Level,
+} from '../coreInterfaces';
+import {
+  CONTENT_LENGTH_HEADER,
+  CONTENT_TYPE_HEADER,
+  getHeader,
+} from '../http/httpHeaders';
+import { NullLogger } from './nullLogger';
+
+export class ApiLogger implements ApiLoggerInterface {
+  private readonly _loggingOptions: LoggingOptions;
+  private readonly _isConfigured: boolean;
+  private readonly _logger: LoggerInterface;
+
+  constructor(loggingOpt: LoggingOptions) {
+    this._loggingOptions = loggingOpt;
+    this._logger = loggingOpt.logger ?? new NullLogger();
+    this._isConfigured = this._logger && this._logger !== new NullLogger();
+  }
+
+  public logRequest(request: HttpRequest): void {
+    if (!this._isConfigured) {
+      return;
+    }
+
+    const logLevel = this._loggingOptions.logLevel ?? Level.Info;
+    const contentTypeHeader = this.getContentType(request.headers);
+    const url = this._loggingOptions.logRequest?.includeQueryInPath
+      ? request.url
+      : this.removeQueryParams(request.url);
+
+    this._logger.log(
+      logLevel,
+      'Request  HttpMethod: ${method} Url: ${url} ContentType: ${contentType}',
+      {
+        method: request.method,
+        url,
+        contentType: contentTypeHeader,
+      }
+    );
+
+    this.applyLogRequestOptions(logLevel, request);
+  }
+
+  public logResponse(response: HttpResponse): void {
+    if (!this._isConfigured) {
+      return;
+    }
+
+    const logLevel = this._loggingOptions.logLevel ?? Level.Info;
+    const contentTypeHeader = this.getContentType(response.headers);
+    const contentLengthHeader = this.getContentLength(response.headers);
+
+    this._logger.log(
+      logLevel,
+      'Response HttpStatusCode: ${statusCode} Length: ${contentLength} ContentType: ${contentType}',
+      {
+        statusCode: response.statusCode,
+        contentLength: contentLengthHeader,
+        contentType: contentTypeHeader,
+      }
+    );
+
+    this.applyLogResponseOptions(logLevel, response);
+  }
+
+  private applyLogRequestOptions(level: Level, request: HttpRequest) {
+    this.applyLogRequestHeaders(
+      level,
+      request,
+      this._loggingOptions.logRequest
+    );
+
+    this.applyLogRequestBody(level, request, this._loggingOptions.logRequest);
+  }
+
+  private applyLogRequestHeaders(
+    level: Level,
+    request: HttpRequest,
+    logRequest?: HttpRequestLoggingOptions
+  ) {
+    if (logRequest) {
+      const { logHeaders, headerToInclude, headerToExclude } = logRequest;
+
+      if (logHeaders) {
+        const headersToLog = this.extractHeadersToLog(
+          request.headers,
+          headerToInclude,
+          headerToExclude
+        );
+
+        this._loggingOptions.logger?.log(level, 'Request Headers: ${headers}', {
```

Comment:
Use the instance of logger stored in this class:
```suggestion
        this._logger.log(level, 'Request Headers: ${headers}', {
```

Reply:
I have fixed it.